src/SecretManager/CommandOutputLogger.cs

```diff
@@ -13,10 +13,12 @@ namespace SecretManager
     public class CommandOutputLogger : ILogger
     {
         private readonly CommandOutputProvider _provider;
+        private readonly bool _isWindows;
 
         public CommandOutputLogger(CommandOutputProvider commandOutputProvider)
         {
             _provider = commandOutputProvider;
+            _isWindows = _provider._runtimeEnv.OperatingSystem == "Windows";
```

Comment:
This looks bad. Just add a property on `CommandOutputProvider` return if the `OperatingSystem == "Windows"`